scheduler/generic_sched.go

```diff
@@ -657,6 +658,21 @@ func (s *GenericScheduler) computePlacements(destructive, place []placementResul
 						"old_alloc_name", oldAllocName, "new_alloc_name", newAllocName)
 				}
 
+				// Are there sticky volumes requested by the task group for the first time? If
+				// yes, make sure the allocation stores their IDs for future reschedules.
+				var newHostVolumeIDs []string
+				for _, v := range tg.Volumes {
+					if v.Sticky {
+						if missing.PreviousAllocation() != nil && len(missing.PreviousAllocation().HostVolumeIDs) > 0 {
+							continue
+						}
+						vol, ok := option.Node.HostVolumes[v.Source]
+						if ok {
+							newHostVolumeIDs = append(newHostVolumeIDs, vol.ID)
+						}
+					}
+				}
```

Comment:
I think we can add the host volume IDs unconditionally at this point? We have a selected alloc-node pair and so any dynamic host volumes the alloc wants _must_ be the correct volume IDs.